providers/amazon/src/airflow/providers/amazon/aws/bundles/s3.py

```diff
@@ -137,10 +137,20 @@ def refresh(self) -> None:
             )
 
     def view_url(self, version: str | None = None) -> str | None:
+        """
+        Return a URL for viewing the DAGs in S3. Currently, versioning is not supported.
+
+        This method is deprecated and will be removed in a future release. Use `view_url_template` instead.
+        """
+        return self.view_url_template()
+
+    def view_url_template(self) -> str | None:
         """Return a URL for viewing the DAGs in S3. Currently, versioning is not supported."""
         if self.version:
             raise AirflowException("S3 url with version is not supported")
-
+        if hasattr(self, "_view_url_template") and self._view_url_template:
+            # Backward compatibility for released Airflow versions
+            return self._view_url_template
```

Comment:
I do not understand this compat. The `_view_url_template` is a new field right? How is it present on older Airflow?

Reply:
Airflow 3.0.2 for example doesn't have the `_view_url_template`. If this provider is released and used in `3.0.2`, it will raise errors